<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>RNA Library Preparation &mdash; Drew Honson 0.0.0 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="RNA Miniprep" href="rnamini.html" />
    <link rel="prev" title="DNA Library Preparation" href="dnalibrary.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            Drew Honson
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Laboratory Protocols</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="dnalibrary.html">DNA Library Preparation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">RNA Library Preparation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#materials">Materials</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#in-house-reagents">In-House Reagents</a></li>
<li class="toctree-l3"><a class="reference internal" href="#commercial-reagents">Commercial Reagents</a></li>
<li class="toctree-l3"><a class="reference internal" href="#plastics-and-beads">Plastics and Beads</a></li>
<li class="toctree-l3"><a class="reference internal" href="#oligonucleotides">Oligonucleotides</a></li>
<li class="toctree-l3"><a class="reference internal" href="#for-library-diagnostics">For Library Diagnostics</a></li>
<li class="toctree-l3"><a class="reference internal" href="#for-final-library-preparation">For Final Library Preparation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#adapter-annealing">Adapter Annealing</a></li>
<li class="toctree-l2"><a class="reference internal" href="#fragmentation-and-end-repair">Fragmentation and End Repair</a></li>
<li class="toctree-l2"><a class="reference internal" href="#rna-ligation">RNA Ligation</a></li>
<li class="toctree-l2"><a class="reference internal" href="#reverse-transcription">Reverse Transcription</a></li>
<li class="toctree-l2"><a class="reference internal" href="#cdna-splint-ligation">cDNA Splint Ligation</a></li>
<li class="toctree-l2"><a class="reference internal" href="#library-amplification">Library Amplification</a></li>
<li class="toctree-l2"><a class="reference internal" href="#library-diagnostics">Library Diagnostics</a></li>
<li class="toctree-l2"><a class="reference internal" href="#final-library-preparation">Final Library Preparation</a></li>
<li class="toctree-l2"><a class="reference internal" href="#protocol-notes">Protocol Notes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#hazards-and-waste-disposal">Hazards and Waste Disposal</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="rnamini.html">RNA Miniprep</a></li>
<li class="toctree-l1"><a class="reference internal" href="silane.html">Silane Nucleic Acid Purification</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Software Packages</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../softwarepackages/rapprobesdemo.html">RAP Probe Generation</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">About</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../aboutme/whoami.html">What’s My Deal?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../aboutme/sitedesign.html">Site Design</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Drew Honson</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">RNA Library Preparation</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/laboratoryprotocols/rnalibrary.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="rna-library-preparation">
<h1>RNA Library Preparation<a class="headerlink" href="#rna-library-preparation" title="Permalink to this heading"></a></h1>
<p>This protocol describes how to generate Illumina sequencing libraries from total RNA. For already fragmented RNA (e.g. from eCLIP) the initial fragmentation step can be omitted. The earliest version of this protocol I can find is from Jesse Engreitz’s 2013 paper <a class="reference external" href="https://doi.org/10.1126/science.1237973">“The Xist lncRNA Exploits Three-Dimensional Genome Architecture to Spread Across the X Chromosome”</a> but it has undergone many improvements over the subsequent years thanks to the work of numerous scientists in the Guttman lab. I’m presenting it here not to claim credit, but rather to increase distribution of the protocol. Additionally, the <a class="reference external" href="https://dhonson-lncrna.github.io/laboratoryprotocols/dnalibrary.html">DNA Library Preparation protocol</a> (which I did write) seems lonely without its RNA partner.</p>
<p>The workflow begins by fragmenting RNA to sizes between 100-1000bp and dephosphorylating the fragments. A common adapter is then ligated to the 3’-end of the RNA to allow for reverse transcription (RT) with a common primer. After RT, the cDNA has an i7 amplifiable 5’-end. The next step is ligate a splint to the 3’-end of the cDNA to permit amplification with an i5 primer. After removing excess splint, the RNA is ready for PCR amplification with Nextera XT primers.</p>
<figure class="align-center" id="id1">
<a class="reference internal image-reference" href="../_images/23.08.11_RNA-Lib-Fig.png"><img alt="Schematic of RNA lib workflow" src="../_images/23.08.11_RNA-Lib-Fig.png" style="width: 600px;" /></a>
<figcaption>
<p><span class="caption-text"><em>Schematic of library prep workflow. i5 is on the left side (green) while i7 is on the right (orange).</em></span><a class="headerlink" href="#id1" title="Permalink to this image"></a></p>
</figcaption>
</figure>
<br><section id="materials">
<h2>Materials<a class="headerlink" href="#materials" title="Permalink to this heading"></a></h2>
<section id="in-house-reagents">
<h3>In-House Reagents<a class="headerlink" href="#in-house-reagents" title="Permalink to this heading"></a></h3>
<p><a class="reference external" href="https://docs.google.com/spreadsheets/d/14t-B8F6yNJ7xcO8kBrjlpoRxs6Zq3iMc/edit?usp=sharing&amp;ouid=110781881215035430006&amp;rtpof=true&amp;sd=true">Buffer calculator</a></p>
<p>10x Annealing Buffer</p>
<blockquote>
<div><p>100mM Tris-HCl, pH 7.5</p>
<p>2M LiCl</p>
</div></blockquote>
<p>GuSCN Lysis Buffer</p>
<blockquote>
<div><p>4M guanidinium thiocyanate (GuSCN)</p>
<p>55mM Tris-HCl, pH 7.5</p>
<p>25mM EDTA</p>
<p>3% Triton X-100</p>
</div></blockquote>
<p>Wash Buffer 2</p>
<blockquote>
<div><p>80% ethanol</p>
<p>100mM NaCl</p>
<p>10mM Tris-HCl, pH7.5</p>
</div></blockquote>
<p>100% ethanol</p>
<p>80% ethanol</p>
<p>500mM EDTA, pH8.0</p>
<p>1M NaOH</p>
<p>1M HCl</p>
<p>DMSO</p>
</section>
<section id="commercial-reagents">
<h3>Commercial Reagents<a class="headerlink" href="#commercial-reagents" title="Permalink to this heading"></a></h3>
<p>NEB 100mM ATP</p>
<p>NEB 10mM dNTP Mix</p>
<p>NEB 2x Instant Sticky-End Master Mix</p>
<p>NEB 2x Q5 High Fidelity Master Mix</p>
<p>NEB 50% PEG8000</p>
<p>NEB Exonuclease I</p>
<p>NEB T4 PNK and 10x buffer</p>
<p>NEB T4 High Concentration RNA Ligase and 10x buffer</p>
<p>ThermoFisher Fast AP and 10x buffer</p>
<p>ThermoFisher Maxima H Minus RT and 5x RT buffer</p>
<blockquote>
<div><p>The exact reverse transcriptase is flexible but it must <strong>lack</strong> RNaseH activity</p>
</div></blockquote>
<p>ThermoFisher Ribolock (or other RNase inhibitor)</p>
<p>Zymo RNA Binding Buffer</p>
<p>Zymo RNA Prep Buffer</p>
</section>
<section id="plastics-and-beads">
<h3>Plastics and Beads<a class="headerlink" href="#plastics-and-beads" title="Permalink to this heading"></a></h3>
<p>SPRI beads (Beckman Coulter or Bulldog Bio)</p>
<p>ThermoFisher Dynabeads MyOne Silane</p>
<p>Zymo IC Columns</p>
<p>Zymo Collection Tubes</p>
</section>
<section id="oligonucleotides">
<h3>Oligonucleotides<a class="headerlink" href="#oligonucleotides" title="Permalink to this heading"></a></h3>
<p>RiL19 (RNA Ligation Adapter):</p>
<blockquote>
<div><p>/5Phos/rArGrArUrCrGrGrArArGrArGrCrGrUrCrGrUrG/ddC</p>
</div></blockquote>
<p>AR17 (RT Primer):</p>
<blockquote>
<div><p>ACACGACGCTCTTCCGA</p>
</div></blockquote>
<p>3Tr3 Splint TOP (cDNA ligation adapter):</p>
<blockquote>
<div><p>CAGACGTGTGCTCTTCCGATCTNNNNNN/3SpC3/</p>
</div></blockquote>
<p>3Tr3 Splint BOT (cDNA ligation adapter):</p>
<blockquote>
<div><p>/5Phos/AGATCGGAAGAGCACACGTCTG</p>
</div></blockquote>
<p>Nextera XT i7 F and i5 R primers, 12.5µM mix (see Protocol Notes)</p>
</section>
<section id="for-library-diagnostics">
<h3>For Library Diagnostics<a class="headerlink" href="#for-library-diagnostics" title="Permalink to this heading"></a></h3>
<p>Qubit HS DNA kit or other high-sensitivity spectrophotometer</p>
<p>TapeStation, Bioanalyzer, or other high sensitivity electrophoresis instrument</p>
</section>
<section id="for-final-library-preparation">
<h3>For Final Library Preparation<a class="headerlink" href="#for-final-library-preparation" title="Permalink to this heading"></a></h3>
<p>1-2% agarose gel</p>
<p>Zymo ADB Buffer</p>
</section>
</section>
<section id="adapter-annealing">
<h2>Adapter Annealing<a class="headerlink" href="#adapter-annealing" title="Permalink to this heading"></a></h2>
<p>3Tr3 splint adds a 3’-overhang for the Nextera Illumina i5 primer set following ligation to cDNA.</p>
<ol class="arabic">
<li><p>3Tr3 Annealing Reaction</p>
<p>18µl 100µM 3Tr3 Splint TOP</p>
<p>18µl 100µM 3Tr3 Splint BOT</p>
<p>4µl 10x annealing buffer</p>
</li>
<li><p>Heat reactions to 95°C on a thermocycler, then ramp down to 25°C, 5°C decrease per 15sec.</p></li>
<li><p>Place the reactions on ice. Adapters are now at 45µM. Take an aliquot of the annealed adapter and dilute to 4.5µM using TE buffer or nuclease-free water. Store annealed adapter at -20°C.</p></li>
</ol>
</section>
<section id="fragmentation-and-end-repair">
<h2>Fragmentation and End Repair<a class="headerlink" href="#fragmentation-and-end-repair" title="Permalink to this heading"></a></h2>
<ol class="arabic">
<li><p>RNA Fragmentation Mix</p>
<p>25ng-1µg RNA</p>
<p>2µl 10x Fast AP Buffer</p>
<p>Water to 18µl</p>
</li>
<li><p>Heat 91°C, 2min30s then cool to 4°C.</p></li>
<li><p>Add 1µl Fast AP and 1µl RNase inhibitor. Incubate 10min, 37°C.</p></li>
<li><p>Dephosphorylation Reaction</p>
<p>20µl Fast AP reaction</p>
<p>3µl 10x T4 PNK Buffer</p>
<p>3µl T4 PNK</p>
<p>4µl water</p>
</li>
<li><p>Incubate 10min, 37°C.</p></li>
<li><p>Column clean the reaction.</p>
<ol class="loweralpha simple">
<li><p>Add 2x volume (60µl) RNA binding buffer.</p></li>
<li><p>Add 1x volume (90µl) 100% ethanol.</p></li>
<li><p>Invert 5-10 times to mix then pop spin briefly.</p></li>
<li><p>Transfer solution to a Zymo IC column. Spin &gt;8000g, 15s.</p></li>
<li><p>Discard flow through.</p></li>
<li><p>Add 400µl RNA Prep Buffer. Spin &gt;8000g, 15s.</p></li>
<li><p>Discard flow through.</p></li>
<li><p>Add 500µl Wash Buffer 2. Spin &gt;8000g, 15s.</p></li>
<li><p>Discard flow through.</p></li>
<li><p>Add 500µl Wash Buffer 2. Spin &gt;8000g, 15s.</p></li>
<li><p>Transfer column to a clean 1.7mL tube.</p></li>
<li><p>Add <strong>12µl</strong> water. Spin 8000-16,000g, 1min.</p></li>
</ol>
</li>
</ol>
</section>
<section id="rna-ligation">
<h2>RNA Ligation<a class="headerlink" href="#rna-ligation" title="Permalink to this heading"></a></h2>
<ol class="arabic">
<li><p>Mix 1.5µl DMSO and 1µl 20µM RNA Ligation Adapter.</p></li>
<li><p>Add 10µl fragmented and end repaired RNA to DMSO/adapter mix.</p></li>
<li><p>Heat 2min, 65°C then cool to 4°C.</p></li>
<li><p>Prepare Ligation Master Mix. After mixing, mix the reaction by flicking the tube until it splits into many droplets. Pop-spin the tube to gather the liquid. Repeat this 4-5 times to thoroughly mix. This step is essential due to the viscosity of PEG8000.</p>
<p>2µl 10x T4 RNA Ligase Buffer</p>
<p>0.2µl 100mM ATP</p>
<p>6µl 50% PEG8000, pre-warmed to 37°C</p>
<p>0.3µl RNase inhibitor</p>
<p>0.3µl T4 High Concentration RNA Ligase</p>
</li>
<li><p>Add 8.8µl Ligation Master Mix to 12.5µl RNA/adapter mix.</p></li>
<li><p>Mix reaction by flicking and spinnning as done with the master mix.</p></li>
<li><p>Add an additional 1µl T4 High Concentration RNA Ligase. Mix reaction by flicking and spinning.</p></li>
<li><p>Incubate 1h15m, 25°C, 1600rpm shaking (1min ON, 5min OFF)</p></li>
<li><p>Clean using Silane beads. For a more detailed description, see <a class="reference external" href="https://dhonson-lncrna.github.io/laboratoryprotocols/silane.html">Silane Nucleic Acid Purification.</a></p>
<ol class="loweralpha simple">
<li><p>Wash 12µl Silane beads per sample 3 times in 500µl GuSCN Lysis Buffer minimum, or 500µl GuSCN Lysis buffer per 120µl beads.</p></li>
<li><p>After removing final wash, resuspend beads in 60µl GuSCN Lysis Buffer per sample.</p></li>
<li><p>Add 60µl beads to each sample. Pipette up and down to mix.</p></li>
<li><p>Incubate 1min, room temperature.</p></li>
<li><p>Add 50µl 100% ethanol. Pipette up and down to mix.</p></li>
<li><p>Incubate 3min, room temperature.</p></li>
<li><p>Place tubes on magnetic rack and wait until solution clears.</p></li>
</ol>
<ol class="loweralpha simple" start="5">
<li><p>Remove supernatant. Wash beads three times with 150µl 80% ethanol if using strip tubes, or 250µl 80% ethanol if using 1.7mL tubes.</p></li>
<li><p>After removing final wash, pop-spin then place the tube back on the magnet. Use a P20 pipette to remove any residual ethanol.</p></li>
<li><p>Dry the beads 5-10min until they look matte.</p></li>
<li><p>Resuspend the beads in <strong>13µl</strong> nuclease-free water to elute.</p></li>
</ol>
</li>
</ol>
</section>
<section id="reverse-transcription">
<h2>Reverse Transcription<a class="headerlink" href="#reverse-transcription" title="Permalink to this heading"></a></h2>
<ol class="arabic">
<li><p>Annealing Reaction</p>
<p>12.5µl RNA sample</p>
<p>1µl 5µM AR17</p>
<p>1µl 10mM dNTPs</p>
</li>
<li><p>Heat 2min, 65°C then cool to 4°C.</p></li>
<li><p>Add 5.5µl RT Master Mix.</p>
<p>4µl 5x Maxima RT Buffer</p>
<p>1µl Maxima RT RNaseH Minus</p>
<p>0.5µl RNase inhibitor</p>
</li>
<li><p>Incubate 30min, 50°C then cool to 4°C.</p></li>
<li><p>Add 4µl Exonuclease I to digest excess AR17. Incubate 15min, 37°C.</p></li>
<li><p>Add 1µl 0.5M EDTA to inactivate Exonuclease I.</p></li>
<li><p>Place on ice for 3min.</p></li>
<li><p>Add 2.5µl 1M NaOH.</p></li>
<li><p>Heat 6min, 80°C to degrade RNA, then cool to 4°C.</p></li>
<li><p>Add 2.5µl 1M HCl to neutralize base.</p></li>
<li><p>Clean using Silane beads.</p></li>
</ol>
<blockquote>
<div><ol class="loweralpha simple">
<li><p>Wash 12µl Silane beads per sample 3 times in 500µl GuSCN Lysis Buffer minimum, or 500µl GuSCN Lysis buffer per 120µl beads.</p></li>
<li><p>After removing final wash, resuspend beads in 90µl GuSCN Lysis Buffer per sample.</p></li>
<li><p>Add 90µl beads to each sample. Pipette up and down to mix.</p></li>
<li><p>Incubate 1min, room temperature.</p></li>
<li><p>Add 75µl 100% ethanol. Pipette up and down to mix.</p></li>
<li><p>Incubate 3min, room temperature.</p></li>
<li><p>Place tubes on magnetic rack and wait until solution clears.</p></li>
</ol>
<ol class="loweralpha simple" start="5">
<li><p>Remove supernatant. Wash beads three times with 150µl 80% ethanol if using strip tubes, or 250µl 80% ethanol if using 1.7mL tubes.</p></li>
<li><p>After removing final wash, pop-spin then place the tube back on the magnet. Use a P20 pipette to remove any residual ethanol.</p></li>
<li><p>Dry the beads 5-10min until they look matte.</p></li>
<li><p>Resuspend the beads in 13µl nuclease-free water to elute.</p></li>
</ol>
</div></blockquote>
</section>
<section id="cdna-splint-ligation">
<h2>cDNA Splint Ligation<a class="headerlink" href="#cdna-splint-ligation" title="Permalink to this heading"></a></h2>
<ol class="arabic">
<li><p>Ligation Reaction</p>
<p>2µl 4.5µM 3Tr3 Splint</p>
<p>13µl cDNA</p>
<p>15µl 2x Instant Sticky-End Master Mix</p>
</li>
<li><p>Incubate 30min, 25°C</p></li>
<li><p>SPRI clean to remove adapter.</p>
<ol class="loweralpha simple">
<li><p>Add 1x volume room-temperature SPRI beads (30µl) to the reaction. Flick and pop-spin the tube 3 times to mix.</p></li>
<li><p>Incubate 9min, room-temperature. Flick and pop-spin every 3min to mix.</p></li>
<li><p>Place tube on magnetic rack and allow beads to fully clear, 2-5min.</p></li>
<li><p>Remove supernatant.</p></li>
<li><p>Add 150µl 80% ethanol. Remove tube from rack, rotate 180°, then place back in rack to pull beads through the ethanol to the opposite wall. Repeat rotation 2-3 times.</p></li>
<li><p>Allow beads to separate completely and remove supernatant.</p></li>
<li><p>Repeat steps e. and f. twice more for a total of 3 washes.</p></li>
<li><p>After removing final ethanol wash, briefly pop-spin the tubes. Place on the magnet and use a P20 pipette to remove any residual ethanol.</p></li>
<li><p>Dry beads 2-5min until they look matte.</p></li>
<li><p>Resuspend beads in <strong>50µl</strong> nuclease-free water to elute. Magnetically pellet the beads and transfer the supernatant to a clean tube.</p></li>
</ol>
</li>
</ol>
</section>
<section id="library-amplification">
<h2>Library Amplification<a class="headerlink" href="#library-amplification" title="Permalink to this heading"></a></h2>
<ol class="arabic">
<li><p>Library PCR Reaction</p>
<p>22.5µl ligated cDNA</p>
<p>2.5µl 12.5µM i7 F x i5 R Broad Primers</p>
<p>25µl 2x Q5 High-Fidelity Master Mix</p>
</li>
<li><p>Thermocycler Settings</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 33%" />
<col style="width: 33%" />
<col style="width: 33%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Temp (°C)</p></th>
<th class="head"><p>Time</p></th>
<th class="head"><p>Cycles</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>98</p></td>
<td><p>15s</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><p>98</p></td>
<td><p>15s</p></td>
<td><p>4</p></td>
</tr>
<tr class="row-even"><td><p>68</p></td>
<td><p>15s</p></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>72</p></td>
<td><p>30s</p></td>
<td></td>
</tr>
<tr class="row-even"><td><p>98</p></td>
<td><p>15s</p></td>
<td><p>3-9 (see Protocol Notes)</p></td>
</tr>
<tr class="row-odd"><td><p>72</p></td>
<td><p>1min</p></td>
<td></td>
</tr>
<tr class="row-even"><td><p>72</p></td>
<td><p>1min</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><p>4</p></td>
<td><p>forever</p></td>
<td></td>
</tr>
</tbody>
</table>
</li>
<li><p>SPRI clean to remove primers.</p>
<ol class="loweralpha simple">
<li><p>Add 1x volume room-temperature SPRI beads (50µl) to the reaction. Flick and pop-spin the tube 3 times to mix.</p></li>
<li><p>Incubate 9min, room-temperature. Flick and pop-spin every 3min to mix.</p></li>
<li><p>Place tube on magnetic rack and allow beads to fully clear, 2-5min.</p></li>
<li><p>Remove supernatant.</p></li>
<li><p>Add 150µl 80% ethanol. Remove tube from rack, rotate 180°, then place back in rack to pull beads through the ethanol to the opposite wall. Repeat rotation 2-3 times.</p></li>
<li><p>Allow beads to separate completely and remove supernatant.</p></li>
<li><p>Repeat steps e. and f. twice more for a total of 3 washes.</p></li>
<li><p>After removing final ethanol wash, briefly pop-spin the tubes. Place on the magnet and use a P20 pipette to remove any residual ethanol.</p></li>
<li><p>Dry beads 2-5min until they look matte.</p></li>
<li><p>Resuspend beads in <strong>13µl</strong> nuclease-free water to elute. Magnetically pellet the beads and transfer the supernatant to a clean tube.</p></li>
</ol>
</li>
</ol>
</section>
<section id="library-diagnostics">
<h2>Library Diagnostics<a class="headerlink" href="#library-diagnostics" title="Permalink to this heading"></a></h2>
<ol class="arabic simple">
<li><p>Measure concentration of library with Qubit High Sensitivity.</p></li>
<li><p>Measure size distribution by loading 1ng library on a HS D1000 TapeStation Tape or a HS DNA Bioanalyzer chip.</p></li>
<li><p>For standard RNA fragmentation, the library should show a unimodal distribution with an average size between 300-500bp and a tail to the right. For methods with high temperature washes (e.g. CLAP) the average sizes may be smaller. Little if any DNA should have sizes greater than 1000bp. See the image below for an example of a typical library.</p></li>
</ol>
<a class="reference internal image-reference" href="../_images/RNA_Library_Trace.png"><img alt="Example RNA library trace" class="align-center" src="../_images/RNA_Library_Trace.png" style="width: 500px;" /></a>
<br></section>
<section id="final-library-preparation">
<h2>Final Library Preparation<a class="headerlink" href="#final-library-preparation" title="Permalink to this heading"></a></h2>
<p>Libraries should always be gel cut before loading on Illumina instruments to avoid primer contamination. Different Illumina instruments and flow cells require different concentrations of libraries, so always check the documentation of your specifc setup before proceeding.</p>
<p>In the future, I will add a more detailed guide on Illumina sequencing, but for the purpose of this protocol I will assume the user has already decided the number of reads they ultimately want for each library.</p>
<ol class="arabic simple">
<li><p>Pool libraries. Calculate the total femtomoles to pool by multiplying the desired final concentration by the final elution volume (13µl). Multiply this number by two to account for loss during clean-up. E.g. for a 1nM final library, multiply 1nM (fmol/µl) by 13µl by 2 to get 26fmol to pool).</p></li>
<li><p>Load pools on a 1-2% agarose gel. Keep one lane empty between ladders and individual pools. Run gel until ladder is clearly separated.</p></li>
<li><p>Using a razor blade, cut libraries between 200-1000bp. Transfer excised agarose to a clean 1.7mL tube.</p></li>
</ol>
<figure class="align-center" id="id2">
<a class="reference internal image-reference" href="../_images/22.10.03_LibPools.png"><img alt="Pre- and post-gel cut libraries" src="../_images/22.10.03_LibPools.png" style="width: 600px;" /></a>
<figcaption>
<p><span class="caption-text"><em>Example libraries before and after gel-cutting.</em></span><a class="headerlink" href="#id2" title="Permalink to this image"></a></p>
</figcaption>
</figure>
<ol class="arabic simple" start="4">
<li><p>Tare a balance with an empty 1.7mL tube. Mass the excised agarose, then add 3µl Zymo ADB Buffer per 1mg agarose gel (e.g. 300µl ABD for a 100mg gel).</p></li>
<li><p>Heat gel at 50°C until fully dissolved, 3-5min.</p></li>
<li><p>Load up to 700µl dissolved gel onto a Zymo IC column in a collection tube. Spin &gt;8000g, 15s. Discard flow through and repeat until entire sample has passed through the column.</p></li>
<li><p>Add 200µl DNA/RNA Wash Buffer. Spin &gt;8000g, 15s.</p></li>
<li><p>Discard flow through.</p></li>
<li><p>Add 200µl DNA/RNA Wash Buffer. Spin &gt;8000g, 1min to dry column.</p></li>
<li><p>Discard collection tube and transfer column to a clean 1.7mL tube.</p></li>
<li><p>Add 13µl nuclease-free water directly to the column. Incubate at room temperature 2-3min.</p></li>
<li><p>Spin 8000-16,000g, 1min to elute.</p></li>
<li><p>Quantify final gel-cut library using Qubit and Tapestation or Bioanalyzer as before.</p></li>
</ol>
</section>
<section id="protocol-notes">
<h2>Protocol Notes<a class="headerlink" href="#protocol-notes" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>The sequences of Illumina’s Nextera XT’s primers are <a class="reference external" href="https://dnatech.genomecenter.ucdavis.edu/wp-content/uploads/2013/06/illumina-adapter-sequences_1000000002694-00.pdf">publicly available.</a> For convenience, I plan to post spreadsheets containing the sequences and adaptors that are compatible with IDT ordering on a GoogleDrive associated with this site.</p></li>
<li><p>The number of cycles will vary dramatically depending on the type of experiment and amount of input DNA. Once a protocol is well-established a user can usually estimate an appropriate cycle number, but in early stages of using the protocol it is best to determine the optimal cycle number empirically. <a class="reference external" href="https://www.lexogen.com/amplification-of-rna-seq-libraries-the-correct-pcr-cycle-number/">This site</a> provides a good explanation of how to use qPCR for this purpose.</p></li>
<li><p>An inexpensive supplier of SPRI beads in the United States is <a class="reference external" href="https://www.bulldog-bio.com/product/cleanngs-beads/">Bulldog Bio.</a> We have not noticed any performance differences between Bulldog Bio’s and Beckman Coulter’s products, but it is always best to check size selection using a DNA ladder when switching to a new product.</p></li>
<li><p>If preparing many libraries simultaneously, it may be more convenient to perform Silane cleans rather than column cleans. Please refer to my Silane Cleaning Protocol for this adaptation.</p></li>
</ul>
</section>
<section id="hazards-and-waste-disposal">
<h2>Hazards and Waste Disposal<a class="headerlink" href="#hazards-and-waste-disposal" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p>Consult the Safety Data Sheet for any unfamiliar reagents and comply with local regulations regarding disposal of hazardous waste.</p></li>
<li><p>Zymo RNA Binding Buffer contains guanidine hydrochloride (GuHCl), a powerful chaotrope. Always wear gloves and a labcoat when handling GuHCl. Do not open solid GuHCl outside of a fume hood and dispose of all liquid and solid waste in dedicated containers.</p></li>
<li><p>Solutions containing GuHCl produce chlorine gas when mixed with bleach or other oxidizers. Be mindful of keeping these reagents separate.</p></li>
<li><p>GuSCN is a powerful chaotrope. Always wear gloves and a labcoat when handling GuSCN. Do not open solid GuSCN outside of a fume hood and dispose of all liquid and solid waste in dedicated containers.</p></li>
<li><p>GuSCN solutions release hydrogen cyanide gas when exposed to acids or oxidizers such as bleach. Always use caution when preparing or disposing of GuSCN solutions.</p></li>
</ul>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="dnalibrary.html" class="btn btn-neutral float-left" title="DNA Library Preparation" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="rnamini.html" class="btn btn-neutral float-right" title="RNA Miniprep" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, Drew Honson.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>
   

  <style>
  /* Sidebar header (and topbar for mobile) */
  .wy-side-nav-search, .wy-nav-top {
    background: #658B69;
  }
  /* Sidebar background color */
  .wy-nav-side {
    background: #d8e3d9;
  }
  /* Sidebar text color */
  .wy-side-nav-search .icon {
    color: #d8e3d9;
  }
  /* Sidebar label color */
  .wy-menu-vertical .caption-text {
    color: #658B69; /* Change this to the desired text color */
  }
  /* Change text color of specific category */
  .wy-menu-vertical li,
  .wy-menu-vertical a {
    color: #000000; /* Change this to the desired text color */
  }
</style>
}


</body>
</html>